src/elements/content-explorer/ContentExplorer.js

```diff
@@ -763,19 +765,28 @@ class ContentExplorer extends Component<Props, State> {
         let itemThumbnails = [];
         if (isGridViewEnabled) {
             const fileAPI = this.api.getFileAPI(false);
-            itemThumbnails = await Promise.all(items.map(item => fileAPI.getThumbnailUrl(item)));
+            itemThumbnails = await Promise.all(
+                items.map(item => {
+                    return item.type === TYPE_FILE ? fileAPI.getThumbnailUrl(item) : null;
+                }),
+            );
         }
 
-        newCollection.items = items.map((obj, index) => {
-            const isSelected = obj.id === selectedId;
-            const currentItem = isSelected ? selectedItem : obj;
+        newCollection.items = items.map((item, index) => {
+            const isSelected = item.id === selectedId;
+            const currentItem = isSelected ? selectedItem : item;
             const thumbnailUrl = isGridViewEnabled ? itemThumbnails[index] : null;
+
             const newItem = {
                 ...currentItem,
                 selected: isSelected,
                 thumbnailUrl,
             };
 
+            if (item.type === TYPE_FILE && thumbnailUrl && !isThumbnailReady(newItem)) {
+                this.attemptThumbnailGeneration(newItem);
```

Comment:
In this case I don't think so.  Some items may have their thumbnails already generated, so we want to set those immediately.  Items that don't already have their thumbnails generated will fall into this if statement.  Since thumbnail generation could take some time, I think we want to do this in the background, without `await`.  Otherwise, all thumbnails would have to wait for all other thumbnails to be generated.